### PR TITLE
Add `agent-receipts show <seq>` subcommand for receipt inspection

### PR DESCRIPTION
**Pull request by ojongerius**
## What

Adds a new `agent-receipts show <seq>` subcommand that retrieves and displays a single receipt by its chain sequence number from the SQLite store. Supports both human-readable tabular output and raw JSON output via `--json` flag.

## Why

Provides operators with a direct way to inspect individual receipts without needing to parse the full chain or write custom queries. The read-only store access makes it safe to run while the daemon is actively writing. This complements the existing `list` and `verify` subcommands.

## Implementation

- **`daemon/internal/showcli/show.go`**: Core logic for the subcommand
  - `Run()` entry point that parses flags (`--db`, `--chain-id`, `--json`) and arguments
  - `resolveChainID()` handles chain selection: auto-detects single chains, requires explicit `--chain-id` for multi-chain stores
  - `writeJSON()` and `writeHuman()` format output (tabular display includes all receipt fields including action-specific metadata like drop counts)
  - Supports environment variables `AGENTRECEIPTS_DB` and `AGENTRECEIPTS_CHAIN_ID`

- **`daemon/internal/showcli/show_test.go`**: Comprehensive test suite (297 lines)
  - `fixtureDB()` helper creates test databases with synthetic receipts
  - 16 test cases covering: successful retrieval, JSON output, not-found errors, chain auto-detection, multi-chain disambiguation, argument validation, environment variable handling, and edge cases (empty store, missing DB, bad flags)

- **`daemon/cmd/agent-receipts/main.go`**: Integrated `show` subcommand into CLI dispatcher

- **`daemon/README.md`**: Added documentation for the new subcommand with usage examples and exit codes

## Exit Codes

| Code | Meaning |
|---|---|
| `0` | Receipt found and printed |
| `1` | No receipt at the requested sequence (or empty store) |
| `2` | Usage error (bad flags, ambiguous chain, unreadable DB) |

## Checklist

- [x] Tests pass for all changed components (16 new unit tests in `show_test.go`)
- [x] No real keys or secrets in the diff
- [x] AGENTS.md updated (README.md structure documented)
- [x] All inputs validated at trust boundaries (sequence parsing, chain resolution, DB access)
- [x] Edge cases tested (empty store, missing DB, multiple chains, invalid arguments)

### CODE REVIEW

File: daemon/internal/showcli/show.go [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+			return v
+		}
+		return fallback
+	}
+
+	fs := flag.NewFlagSet("show", flag.ContinueOnError)
+	fs.SetOutput(stderr)
+	fs.Usage = func() {
+		fmt.Fprintln(stderr, "Usage: agent-receipts show <seq> [flags]")
+		fmt.Fprintln(stderr, "\nPrint the full fields of the receipt at chain sequence <seq>.")
```
[the remark this answers is not in the document]
Comment (ojongerius) [reply daemon/internal/showcli/show.go:56]: Fixed in 48f1a21. The parser now loops — each pass calls `fs.Parse`, peels off one positional into `rest`, and re-runs with the remaining args, so `<seq>` and flags may appear in any order. `TestRun_FlagsAfterSeqAreParsed` covers the documented `show 42 --json --chain-id ...` ordering.

File: daemon/README.md [hunk 2]
```diff
@@ -129,6 +129,34 @@ Exit codes are stable for scripting:
...
 | `2` | Usage error (bad flags, missing key file, unreadable DB) |
 
+## Read interface: `agent-receipts show <seq>`
+
+```sh
+agent-receipts show 42
+# or, against a multi-chain store / explicit DB:
+agent-receipts show 42 --chain-id default --db /var/lib/agentreceipts/receipts.db
+# raw receipt JSON:
+agent-receipts show 42 --json
```
[the remark this answers is not in the document]
Comment (ojongerius) [reply daemon/README.md:168]: The implementation now supports flags after `<seq>` (fixed in 48f1a21), so the documented examples are correct as written.

File: sdk/go/store/store.go [hunk 3]
```diff
@@ -44,7 +44,9 @@ type ReceiptStore interface {
 	Insert(r receipt.AgentReceipt, receiptHash string) error
 	GetByID(id string) (*receipt.AgentReceipt, error)
 	GetChain(chainID string) ([]receipt.AgentReceipt, error)
+	GetByChainSequence(chainID string, sequence int) (*receipt.AgentReceipt, error)
 	GetChainTail(chainID string) (sequence int64, receiptHash string, found bool, err error)
+	DistinctChainIDs() ([]string, error)
 	QueryReceipts(q Query) ([]receipt.AgentReceipt, error)
```
[the remark this answers is not in the document]
Comment (ojongerius) [reply sdk/go/store/store.go:48]: Fixed in 2d76dbf — `GetByChainSequence` and `DistinctChainIDs` are removed from `ReceiptStore`. They live only on `*Store`. `showcli` already used `*store.Store` directly so no callers needed updating.

File: sdk/go/store/store.go [hunk 4]
```diff
@@ -366,6 +368,49 @@ func (s *Store) GetChain(chainID string) ([]receipt.AgentReceipt, error) {
...
+	var rJSON string
+	err := s.db.QueryRow(
+		"SELECT receipt_json FROM receipts WHERE chain_id = ? AND sequence = ?",
+		chainID, sequence,
+	).Scan(&rJSON)
+	if err == sql.ErrNoRows {
+		return nil, nil
+	}
+	if err != nil {
+		return nil, err
```
[the remark this answers is not in the document]
Comment (ojongerius) [reply sdk/go/store/store.go]: Fixed in 2d76dbf — DB errors are now wrapped with `fmt.Errorf("get receipt (chain_id=%s seq=%d): %w", chainID, sequence, err)`, matching the `GetChainTail` pattern.

File: sdk/go/store/store.go [hunk 5]
```diff
@@ -366,6 +368,49 @@ func (s *Store) GetChain(chainID string) ([]receipt.AgentReceipt, error) {
...
+	defer rows.Close()
+	var chains []string
+	for rows.Next() {
+		var id string
+		if err := rows.Scan(&id); err != nil {
+			return nil, err
+		}
+		chains = append(chains, id)
+	}
+	return chains, rows.Err()
```
[the remark this answers is not in the document]
Comment (ojongerius) [reply sdk/go/store/store.go]: Fixed in 2d76dbf — query error, scan error, and `rows.Err()` are all wrapped now (`"enumerate chain ids"`, `"scan chain id"`, `"iterate chain ids"`).

File: sdk/go/store/store.go [hunk 6]
```diff
@@ -44,7 +44,9 @@ type ReceiptStore interface {
 	Insert(r receipt.AgentReceipt, receiptHash string) error
 	GetByID(id string) (*receipt.AgentReceipt, error)
 	GetChain(chainID string) ([]receipt.AgentReceipt, error)
+	GetByChainSequence(chainID string, sequence int) (*receipt.AgentReceipt, error)
 	GetChainTail(chainID string) (sequence int64, receiptHash string, found bool, err error)
+	DistinctChainIDs() ([]string, error)
```
[the remark this answers is not in the document]
Comment (ojongerius) [reply sdk/go/store/store.go]: Verified — all `mcp-proxy` tests pass against this change. The repo-root `go.work` workspace wires `mcp-proxy` to the in-tree `sdk/go`, so the pre-push hook already caught any breakage before this landed.

File: daemon/internal/showcli/show.go [hunk 7]
```diff
@@ -0,0 +1,234 @@
...
+		fmt.Fprintln(stderr, "\nFlags:")
+		fs.PrintDefaults()
+	}
+	dbPath := fs.String("db", envOr("AGENTRECEIPTS_DB", daemon.DefaultDBPath()), "SQLite receipt-store path (env: AGENTRECEIPTS_DB)")
+	// Empty default means "auto-detect": use the sole chain when there is
+	// exactly one, otherwise require the operator to disambiguate.
+	chainID := fs.String("chain-id", envLookup("AGENTRECEIPTS_CHAIN_ID"), "Chain id to read from (env: AGENTRECEIPTS_CHAIN_ID); required only when the store holds more than one chain")
+	asJSON := fs.Bool("json", false, "Output the raw receipt JSON instead of human-readable text")
```
[the remark this answers is not in the document]
Comment (ojongerius) [reply daemon/internal/showcli/show.go]: Fixed in e451ac0 — flag help updated to "pretty-printed JSON". A raw-bytes API on `GetByChainSequence` is a valid longer-term addition, but out of scope here: the daemon exclusively uses `Insert` (not `InsertRaw`), so re-encoding is faithful for the receipts `show` will read in practice.

File: daemon/internal/showcli/show.go [hunk 8]
```diff
@@ -0,0 +1,234 @@
...
+			fmt.Fprintf(stderr, "  %s\n", c)
+		}
+		return "", ExitUsageError
+	}
+}
+
+func writeJSON(stdout, stderr io.Writer, r *receipt.AgentReceipt) int {
+	enc := json.NewEncoder(stdout)
+	enc.SetIndent("", "  ")
+	if err := enc.Encode(r); err != nil {
```
[the remark this answers is not in the document]
Comment (ojongerius) [reply daemon/internal/showcli/show.go:163]: Fixed in e451ac0 — `writeJSON` now checks `syscall.EPIPE` and `io.ErrClosedPipe` and returns `ExitOK`, consistent with `writeHuman`/`exitFromFlush`.

File: daemon/README.md [hunk 9]
```diff
@@ -158,6 +158,34 @@ Exit codes are stable for scripting:
 | `1` | Chain failed verification (output lists per-receipt status) |
 | `2` | Usage error (bad flags, missing key file, unreadable DB) |
 
+## Read interface: `agent-receipts show <seq>`
+
+```sh
+agent-receipts show 42
+# or, against a multi-chain store / explicit DB:
+agent-receipts show 42 --chain-id default --db /var/lib/agentreceipts/receipts.db
+# raw receipt JSON:
```
[the remark this answers is not in the document]
Comment (ojongerius) [reply daemon/README.md]: Fixed in e451ac0 — updated to "pretty-printed JSON".

File: sdk/go/store/store.go [hunk 10]
```diff
@@ -366,6 +366,52 @@ func (s *Store) GetChain(chainID string) ([]receipt.AgentReceipt, error) {
...
+		return nil, nil
+	}
+	if err != nil {
+		return nil, fmt.Errorf("get receipt (chain_id=%s seq=%d): %w", chainID, sequence, err)
+	}
+	var r receipt.AgentReceipt
+	if err := json.Unmarshal([]byte(rJSON), &r); err != nil {
+		return nil, fmt.Errorf("corrupt receipt (chain_id=%s seq=%d): %w", chainID, sequence, err)
+	}
+	return &r, nil
```
[the remark this answers is not in the document]
Comment (ojongerius) [reply sdk/go/store/store.go:389]: Agreed this would be valuable for auditor tooling that needs byte-exact fidelity (e.g. re-verifying signatures against the original signed bytes). Not adding it here — `show` targets operator inspection of daemon-written receipts, which always go through `Insert`. Tracking as a follow-up for when collector-side inspection is needed.
